streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java

```diff
@@ -137,30 +141,46 @@
      * @param interval the time interval between punctuations
      */
     @Deprecated
-    void schedule(long interval);
+    void schedule(final long interval);
 
     /**
-     * Forwards a key/value pair to the downstream processors
+     * Forwards a key/value pair to all downstream processors.
+     * Used the input record's timestamp as timestamp for the output record.
+     *
      * @param key key
      * @param value value
      */
-    <K, V> void forward(K key, V value);
+    <K, V> void forward(final K key, final V value);
+
+    /**
+     * Forwards a key/value pair to the specified downstream processors.
+     * Can be used to set the timestamp of the output record.
+     *
+     * @param key key
+     * @param value value
+     * @param to the options to use when forwarding
+     */
+    <K, V> void forward(final K key, final V value, final To to);
 
     /**
      * Forwards a key/value pair to one of the downstream processors designated by childIndex
      * @param key key
      * @param value value
      * @param childIndex index in list of children of this node
+     * @deprecated please use {@link #forward(Object, Object, To)} instead
```

Comment:
nit:`#forward(K, V, To)`

Reply:
JavaDocs doesn't support generics and falls back to `Object` :(